File: web-ui/src/displays/components/Spinner.js
import React from "react";
import { withStyles } from '@material-ui/core/styles';
import AbstractSpinner from "../../../gen/displays/components/AbstractSpinner";
import SpinnerNotifier from "../../../gen/displays/notifiers/SpinnerNotifier";
import SpinnerRequester from "../../../gen/displays/requesters/SpinnerRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';

const styles = theme => ({});

class Spinner extends AbstractSpinner {

	constructor(props) {
		super(props);
		this.notifier = new SpinnerNotifier(this);
		this.requester = new SpinnerRequester(this);
	};

	refreshLoading = (value) => {
	};
}

export default withStyles(styles, { withTheme: true })(Spinner);
DisplayFactory.register("Spinner", withStyles(styles, { withTheme: true })(Spinner));

File: framework/ui/src/io/intino/alexandria/ui/displays/components/Spinner.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.notifiers.SpinnerNotifier;

public class Spinner<DN extends SpinnerNotifier, B extends Box> extends AbstractSpinner<B> {

    public Spinner(B box) {
        super(box);
    }

    public void showLoading() {
        notifier.refreshLoading(true);
    }

    public void hideLoading() {
        notifier.refreshLoading(false);
    }

}